components/action-mgt/org.wso2.carbon.identity.action.management/src/main/java/org/wso2/carbon/identity/action/management/util/ActionValidator.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.action.management.util;
+
+import org.apache.commons.lang.StringUtils;
+import org.wso2.carbon.identity.action.management.constant.ActionMgtConstants;
+import org.wso2.carbon.identity.action.management.exception.ActionMgtClientException;
+
+import java.util.regex.Pattern;
+
+/**
+ * Action validator class.
+ */
+public class ActionValidator {
+
+    public static final String ACTION_NAME_REGEX = "^[a-zA-Z0-9-_][a-zA-Z0-9-_ ]*[a-zA-Z0-9-_]$";
+    public static final String ENDPOINT_URI_REGEX = "^https://[^\\s/$.?#]\\S*";
+    // According to RFC 9910 a header name must contain only alphanumeric characters, period (.) and hyphen (-),
+    // and should start with an alphanumeric character.
+    public static final String HEADER_REGEX = "^[a-zA-Z0-9][a-zA-Z0-9-.]+$";
+
+    /**
+     * Validate whether required fields exist.
+     *
+     * @param fieldValue Field value.
+     * @throws ActionMgtClientException if the provided field is empty.
+     */
+    public void isBlank(String fieldName, String fieldValue) throws ActionMgtClientException {
```

Comment:
When you have something starting with "is" the expectation is it will return a boolean.

If not better start as 'validateForBlank' or may be you can look at having a one validateField method and validate various aspects and return or thorow

Reply:
This is updated with 4cd2b36e177012dc1fb1909deac8b37495c62a7c